packages/in_app_purchase/in_app_purchase_storekit/shared/in_app_purchase_storekit.podspec

```diff
@@ -18,7 +18,9 @@ Downloaded by pub (not CocoaPods).
   s.documentation_url = 'https://pub.dev/packages/in_app_purchase'
   s.source_files = 'Classes/**/*'
   s.public_header_files = 'Classes/**/*.h'
-  s.dependency 'Flutter'
-  s.platform = :ios, '9.0'
+  s.ios.dependency 'Flutter'
+  s.osx.dependency 'FlutterMacOS'
+  s.ios.deployment_target = '9.0'
+  s.osx.deployment_target = '10.15'
```

Comment:
As far as I can tell, users will get a compile-time build error as was also the root cause of test failures 3 and 4 from [this comment](https://github.com/flutter/plugins/pull/6517#issuecomment-1262937326) prompting them to increase their macOS deployment target to 10.15 or higher. Because of those build failures we also had to change the tool (see [this separate PR](https://github.com/flutter/plugins/pull/6605)).

IIRC users of [`cloud_firestore`](https://pub.dev/packages/cloud_firestore) commonly encounter those build failures when the underlying Firebase SDK has a major version bump, dropping support for older iOS versions. As a plugin client, i.e. user, of [`cloud_firestore`](https://pub.dev/packages/cloud_firestore) myself, I remember having to bump my deployment target a couple of times because of this.

Reply:
@a-wallen As mentioned above, it's not uncommon in the ecosystem for plugins to have higher minimum requirements than the minimum that Flutter itself supports. On Android, for instance, any platform view-based plugin requires 19+ or 20+ (depending on the mode).

If it turns out there are a significant number of people who really don't want to drop pre-10.15 support in order to use this plugin, we could do a follow-up to make the plugin "support" earlier versions but no-op all the methods on those versions (we have some precedent for that approach in some of our iOS plugins). I'd rather not do it if there's not demonstrated demand for it though, since having an app that uses the store but has a viable mode without the store present seems pretty niche, and it adds some annoying boilerplate throughout the plugin.

Reply:
Thanks for the explanations!